buildpacks/maven/src/main.rs

```diff
@@ -0,0 +1,290 @@
+                }
+
+                log_info(format!("Selected Maven version: {}", &version));
+
+                let maven_layer = context
+                    .buildpack_descriptor
+                    .metadata
+                    .tarballs
+                    .get(&version)
+                    .cloned()
+                    .ok_or_else(|| {
+                        MavenBuildpackError::UnsupportedMavenVersion(version.clone()).into()
+                    })
+                    .and_then(|tarball| {
+                        context.handle_layer(layer_name!("maven"), MavenLayer { tarball })
+                    })?;
+
+                log_info(format!("Successfully installed Apache Maven {}", &version));
+
+                (
+                    PathBuf::from("mvn"),
+                    maven_layer.env.apply(
+                        Scope::Build,
+                        &maven_repository_layer
+                            .env
+                            .apply(Scope::Build, &Env::from_current()),
+                    ),
+                )
+            }
+        };
+
+        let maven_goals = context
+            .platform
+            .env()
+            .get("MAVEN_CUSTOM_GOALS")
+            .map_or_else(
+                || Ok(default_maven_goals()),
+                |maven_custom_goals_string| {
+                    shell_words::split(&maven_custom_goals_string.to_string_lossy())
+                        .map_err(MavenBuildpackError::CannotSplitMavenCustomGoals)
+                },
+            )?;
+
+        let mut maven_options = context
+            .platform
+            .env()
+            .get("MAVEN_CUSTOM_OPTS")
+            .map_or_else(
+                || Ok(default_maven_opts()),
+                |maven_custom_opts_string| {
+                    // Since this is a single environment variable, when users want to add multiple
+                    // options, they will expect them to be split like a UNIX shell would. This means
+                    // we need to support proper escaping for options that contain spaces.
+                    shell_words::split(&maven_custom_opts_string.to_string_lossy())
+                        .map_err(MavenBuildpackError::CannotSplitMavenCustomOpts)
+                },
+            )?;
+
+        let settings_xml_path = resolve_settings_xml_path(&context.app_dir, context.platform.env())
+            .map_err(MavenBuildpackError::SettingsError)?;
+
+        if let Some(settings_xml_path) = settings_xml_path {
+            maven_options.push(String::from("-s"));
+            maven_options.push(settings_xml_path.to_string_lossy().to_string());
+        }
+
+        // We need to set some options that relate to buildpack implementation internals. Those
+        // options must not be overridden by the user via MAVEN_CUSTOM_OPTS for the buildpack to
+        // work correctly. We also don't want to show them when we log the Maven command we're
```

Comment:
To avoid confusion, we won't show the users the logs of the Maven command we are running.